rfc/0054-sdk3-management-apis.md

```diff
@@ -2986,13 +2986,13 @@ It is recommended to try to parse the first form and fallback to the second so t
 
 ```
 public interface ICollectionManager{
-    ScopeSpec GetScope(string scopeName, GetScopeOptions options);
-
     Iterable<ScopeSpec> GetAllScopes(GetAllScopesOptions options);
 
-    void CreateCollection(CollectionSpec collection, CreateCollectionOptions options);
+    void CreateCollection(String scopeName, String collectionName, CreateCollectionSettings settings, CreateCollectionOptions options);
```

Comment:
Should the settings here be an attribute of  `CreateCollectionOptions`, considering that they are optional? I could be wrong, but I think in all other operations any optional parameters are part of the options (i.e. the only optional parameter are the options)

Reply:
Personally I argue that when creating a collection you are giving the definition of that collection - though it's fine for that to be the default definition - so it's not an optional parameter.
There's some useful prior art.  Once is that when creating a document you always have to give the content of the document, even though strictly speaking that's optional - valid for a document to be empty/default.
Second is the bucket management API, with createBucket taking a BucketSettings.  (Should have been CreateBucketSettings though, IMO)

Reply:
Ok, that makes sense. I think one difference with bucket management is that there the bucket name is part of the settings, so you can't not provide the bucket settings. Here it is described by the RFC as optional and you are able to simply do `create_collection(scope_name, coll_name)` which just uses the default settings. I guess this is more of an argument for having scopeName and collectionName be part of CreateCollectionSettings though

Reply:
That is true.  I think either way is reasonable.  Having them this way does make the create method nicely consistent with the update and delete methods.  But you're right, it does deviate from the BucketManager model - though I feel that in several ways we didn't nail the BucketManager API, and we can see this new CollectionManager API as an attempt to correct that and setup a better model for the future.
But really, I don't think I have a strong opinion either way.